libs/ui-components/src/components/Device/DeviceDetails/DeviceAliasEdit.tsx

```diff
@@ -0,0 +1,171 @@
+import { getErrorMessage } from '../../../utils/error';
+import RichValidationTextField from '../../form/RichValidationTextField';
+import { getLabelValueValidations, validKubernetesLabelValue } from '../../form/validations';
+
+type DeviceAliasEditProps = { deviceId: string; alias?: string; onAliasEdited: VoidFunction };
+type DeviceAliasEditValues = { alias: string };
+
+const DeviceAliasInputField = ({
+  alias,
+  textInputRef,
+  toggleIsEditing,
+  onSubmit,
+}: {
+  alias?: string;
+  textInputRef: React.RefObject<HTMLInputElement>;
+  toggleIsEditing: VoidFunction;
+  onSubmit: ({ alias }: DeviceAliasEditValues) => void;
+}) => {
+  const { t } = useTranslation();
+
+  const [{ value }, , { setTouched, setValue }] = useField<string>('alias');
+
+  const onAliasKeyDown = React.useCallback(
+    async (e: React.KeyboardEvent<HTMLInputElement>) => {
+      if (e.key === 'Enter' || e.key === 'Tab') {
+        e.preventDefault();
+        const newErrors = await setTouched(true, true);
+        if (Object.keys(newErrors || {}).length === 0) {
+          void onSubmit({ alias: (e.target as HTMLInputElement).value });
+        }
+      } else if (e.key === 'Escape') {
+        e.preventDefault();
+        setValue(alias || t('Untitled'));
+        toggleIsEditing();
+      }
+    },
+    [t, onSubmit, alias, setValue, setTouched, toggleIsEditing],
+  );
+
+  return (
+    <RichValidationTextField
+      fieldName="alias"
+      ref={textInputRef}
+      validations={getLabelValueValidations(t)}
+      placeholder={alias || t('Untitled')}
+      onKeyDown={onAliasKeyDown}
+      onBlur={async () => {
+        const newErrors = await setTouched(true, true);
+        if (Object.keys(newErrors || {}).length === 0) {
+          void onSubmit({ alias: value });
+        }
+      }}
+    />
+  );
+};
+
+const DeviceAliasEdit = ({ deviceId, alias: originalAlias = '', onAliasEdited }: DeviceAliasEditProps) => {
+  const { t } = useTranslation();
+  const { patch } = useFetch();
+
+  const textInputRef = React.useRef<HTMLInputElement>(null);
+
+  const [isEditing, setIsEditing] = React.useState<boolean>();
+  const [isSubmitting, setIsSubmitting] = React.useState<boolean>();
+  const [submitError, setSubmitError] = React.useState<string>();
+
+  const toggleIsEditing = React.useCallback(() => {
+    setIsEditing((wasEditing) => !wasEditing);
+  }, []);
+
+  React.useEffect(() => {
+    // Focus the field after it became editable - before that, the input is not rendered
+    if (isEditing) {
+      textInputRef?.current?.focus();
+    }
+  }, [isEditing]);
+
+  const onSubmit = React.useCallback(
+    async ({ alias }: DeviceAliasEditValues) => {
+      if (alias === originalAlias) {
+        toggleIsEditing();
+        return;
+      }
+      const req: PatchRequest = alias
+        ? [
+            {
+              path: '/metadata/labels/alias',
+              op: 'replace',
+              value: alias,
```

Comment:
I think this will fail if device's labels (or alias label) is undefined

Reply:
I wanted to prevent having to pass the existing labels, and in my testing, labels were always initialized, but I imagine there could be ways to have the labels removed and then it would fail as you mention.

I'll take a look at how to improve this.

Reply:
I couldn't reproduce it, as I only saw that labels are cleared when devices are decommissioned (but those cannot be edited further).
But to be safe, I passed an extra prop to determine when the operation should be `replace` or `add`

Reply:
Last commit is amended. Now the PATCH path would be `/metadata.labels` when no labels were previously defined.